### ISSUE TITLE
Release 1.19.0 - 7 December 2020

### ISSUE THREAD

**Issue by timgl** [PostHog/posthog#2523]
Karl / Paolo
- Player: Solve UX issues (new tab opening): #2548, release
- Track metrics & action #2560
- Persons page new UX #2353
- Sessions page new UX #2352 (but let's figure out a more interactive design)
- Improving sessions page filters (e.g. being able to filter by cohorts, events, session length etc - same as in insights + a few things). Related #2130
- Get billing live, test usage based pricing
- #2367 / #2336 to increase internal sign ups & indirectly retention.
- Data instrumentation

Eric / Tim
- ~Get lifecycle reviewed and merged~
- Improve rentention performance following #2578 
- Rolling retention + #2228 
- ~Retention people breakdown #2562~
- ~Investigate and either fix webhooks on CH OR build as plugins (#2182)~
- refactor the insight filtering to use the filter class pattern in #2534 (in progress)
- ~Fix array filtering #2301~
- Funnel and insight history UX improvements #2524 (in progress)
- Write plugin to push our own users into orbit.love and hubspot
- Cache clickhouse queries (open PR, waiting on stability)

Marius & Michael:
- Get plugins live on EE!
  - Event ingestion (capture_event with all its postgres checks) inside plugin-server
    - Integrate Kafka and plugin-server
  - Tests to verify that nodejs and python ingestions do exactly the same thing
  - ~Tests inside posthog-plugin-server for existing functionality~
  - ~Multithreaded server (with piscina - spawns workers that will handle incoming messages)~
  - Plugins that span teams/organisations.
  - Upgrade plugins inside the posthog app
- ~Instrumentation of plugins (installs/uninstalls)~
  
James G
- Support/Sentry hero
- Move to AWS
- Staging instance + Postgres only instance with high volume?

---

Ignore, from me:

- Improve webpack build speed #2045
- Clearly show what elements we're censoring session recording #2258
- Make the experience for high volume users using postgres feel not buggy

**Comment by EDsCODE** [comment]
Top of mind todos for me:
- get lifecycle reviewed and merged
- carry over on retention epic that needs to be finished (rolling retention is one of the last points)
- add caching to clickhouse insights
- refactor the insight filtering to use the filter class pattern in #2534 (once it's merged successfully)
Possible:
- staging instance

**Comment by mariusandra** [comment]
For "Team Plugin" there's a lot still to do. (@Twixes feel free to suggest other things)

- Event ingestion (`capture_event` with all its postgres checks) inside plugin-server
- Tests to verify that nodejs and python ingestions do exactly the same thing
- Tests inside posthog-plugin-server for existing functionality
- Multithreaded server (with [piscina](https://github.com/piscinajs/piscina ) - spawns workers that will handle incoming messages)
- Scheduled plugins (process running on plugin-server, use a distributed lock with [node-redlock](https://github.com/mike-marcacci/node-redlock) and store the schedule in postgresql)
- Integrate Kafka and plugin-server
- HTTP plugins (webhooks)
- Plugins that span teams/organisations.
- Upgrade plugins inside the posthog app
- Stretch goal: Interface to edit plugin JS in app

**Comment by macobo** [comment]
Roughly in order of priority

### Session recording

- Player: Solve UX issues (new tab opening): https://github.com/PostHog/posthog/issues/2548, release
- Make recording more reliable
  - Track metrics https://github.com/PostHog/posthog/issues/2560
  - Based on metrics, either bump priority of go-based API endpoint, compression, retries or smth else.

After this I think the bottleneck of session recording won't be in the player itself, but instead in:

### Sessions / Persons

Laying the groundwork for a better 'broader' player/sessions UX, tying it together with the rest of the app.

- Persons page new UX https://github.com/PostHog/posthog/issues/2353
- Sessions page new UX https://github.com/PostHog/posthog/issues/2352
- Improving sessions page filters (e.g. being able to filter by cohorts, events, session length etc - same as in insights + a few things). Related https://github.com/PostHog/posthog/issues/2130, full issue not yet written down.

### Session recording - nice-to-haves

I don't feel too strongly on these - if swap out for a different project that would be also ok.

- Player: Better indicate cencored items: https://github.com/PostHog/posthog/issues/2258
- Player: Make progress bar great: https://github.com/PostHog/posthog/issues/2384 
- Saving session recordings to S3: https://github.com/PostHog/posthog/issues/2142
- Being able to record only specific cohorts: https://github.com/PostHog/posthog/issues/2389
- Player: Include more device information

### Misc nice-to-haves
- Posthog-js weirdness https://github.com/PostHog/posthog-js/issues/137 https://github.com/PostHog/posthog-js/issues/138 if working on session recording internals
- Adding more browsers under test for posthog-js

**Comment by paolodamico** [comment]
### General
- **Data instrumentation**. I think we should spend a lot of time making sure we're properly tracking events all throughout PostHog (both making sure that all key elements have appropriate `data-attr`s and that we're sending server events for every key feature) [e.g. we have almost no data on the toolbar, or we don't have certainty on how many actions are being created, ...]. This may also be stuff not specifically related to a code change, but making sure our own actions, dashboards and cohorts are well instrumented too.
   _Rationale:_ We can't improve stuff if we don't know what's working or not, or where can we find shortcomings.
- **Retention Epic** (#2228), particularly #2562.
   _Rationale:_  Dogfooding. This is currently blocking us from doing deeper dives in retention to better work on improving it.
- Echo @macobo on **Sessions & Persons** (rationale on #2366)
- Improve the **sign up & onboarding experience** (#1504, #1547, #1408, #2091). We already have some proposed designs from a couple of months ago, but these need to change as a lot has changed since.
   _Rationale:_  We know our signup & onboarding experience isn't great (from user feedback, and the [significant drop](https://app.posthog.com/insights?insight=FUNNELS&properties=%5B%5D&actions=%5B%7B%22id%22%3A245%2C%22type%22%3A%22actions%22%2C%22order%22%3A0%2C%22name%22%3A%22Signup%20pageview%22%7D%5D&events=%5B%7B%22id%22%3A%22user%20signed%20up%22%2C%22type%22%3A%22events%22%2C%22order%22%3A1%2C%22name%22%3A%22user%20signed%20up%22%7D%5D&new_entity=%5B%5D&date_from=-30d) in the sign up funnel). This will have a significant impact in our WAUs, likely retention, and has more impact now that we're doing paid acquisition.

#### Bug fixes to prioritize
- Fix webhooks on CH (#2182).
   _Rationale:_  We know this is a feature that some power users utilize and that they're directly affected.
- Bug #2301. Rationale: part of core functionality, without it the experimentation power of feature flags essentially disappears (without being able to measure, there's no point in A/B testing).


### Myself
- **Billing stuff**. Test the new tiered pricing (https://github.com/PostHog/ops/issues/67), YC deal (https://github.com/PostHog/ops/issues/70), billing handbook.
    _Rationale:_ This will help validate our hypotheses and data (make sure we're actually solving a problem worth paying for), will reduce the noise from non-serious users (allowing us to weight feedback better), will make us look a lot more professional, and we can start learning on pricing and get a grip on our unit economics. This should also help us address issues around confusion with multiple plans (from conversation with @jamesefhawkins).
- **Benchmarking**. Dive into the products of other product analytics tools to understand what they're doing right (or wrong) that we can leverage to make our product better. This is also related to watching more of our own session recordings.
    _Rationale:_ From customer calls we have obtained specific UX improvements from people that have used other tools before, we will obtain some very useful insights to make our product even better.
- #2367 / #2336 to increase **internal sign ups** & indirectly retention.
    _Rationale:_ hypothesis is that users whose team members are on PostHog retain better.
- More retention

#### Stretch goals
- Fixes & improvements to **new navigation** (#2545, #2541, #2553).
    _Rationale:_ Will enable us to release the new navigation that should help with overall discoverability issues and indirectly squashes a few bugs from the previous navigation bar.
- Fix weekly email reports. Rationale: hypothesis is that this will help retention by driving users back to the product with useful & actionable stuff.

**Comment by Twixes** [comment]
I definitely want to ship specific event ingestion goals, Marius already laid out a lot that needs to be done in that space, but here are two things most immediate for me:
1. Plugin-processed event → ClickHouse pipeline in DB (via Kafka).
2. HTTP capture request → plugin processing pipeline (PR open: PostHog/posthog-plugin-server#25).

**Comment by Twixes** [comment]
Fin.
